lib/cli/src/Cardano/CLI.hs

```diff
@@ -587,18 +611,58 @@ newCliKeyScheme = \case
     ledgerKeyFromSeed = Icarus.getKey
         . flip Icarus.generateKeyFromHardwareLedger pass
 
+    toPub (AXPrv xprv) = return . AXPub . CC.toXPub $ xprv
+    toPub (AXPub _) = Left "Input is already a public key."
+
     derive
         :: CC.DerivationScheme
-        -> XPrv
+        -> XPrvOrXPub
         -> DerivationIndex
-        -> Either String XPrv
-    derive scheme1Or2 k i =
-        return
+        -> Either String XPrvOrXPub
+    derive scheme1Or2 (AXPrv k) i =
+        return . AXPrv
             $ CC.deriveXPrv
                 scheme1Or2
                 pass
                 k
                 (unDerivationIndex i)
+    derive scheme1Or2 (AXPub k) i =
+        maybe (Left err) (return . AXPub)
+            $ CC.deriveXPub
+                scheme1Or2
+                k
+                (unDerivationIndex i)
+      where
+        err = mconcat
+            [ T.unpack (toText i)
+            , " is a hardened index. Public key derivation is only possible for"
+            , " soft indices. \nIf the index is correct, please use the "
+            , " corresponding private key as input."
+            ] -- TODO: Test
+
+    insp (AXPrv key) = do
+        let bytes = CC.unXPrv key
+        let (xprv, rest) = BS.splitAt 64 bytes
+        let (_pub, cc) = BS.splitAt 32 rest
+        let encodeToHex = T.pack . B8.unpack . hex
+        return $ mconcat
+            [ "extended private key: "
+            , encodeToHex xprv
+            , "\n"
+            , "chain code: "
+            , encodeToHex cc
+            ]
+    insp (AXPub key) = do
+        let bytes = CC.unXPub key
+        let (xpub, cc) = BS.splitAt 32 bytes
```

Comment:
Need to check that this is right.

Reply:
Yes https://github.com/input-output-hk/cardano-crypto/blob/2547ad1e80aeabca2899951601079408becbc92c/src/Cardano/Crypto/Wallet.hs#L144-L150